doc/walkthrough.rst

```diff
@@ -0,0 +1,109 @@
+Clusters are groupings of hosts. These hosts are expected to be similar to each other in functionality. In other words,
+the configurations of hosts in a cluster should not differ. While the functionality provided by the hosts may differ
+the system itself should not. Take OpenShift nodes as an example. Some nodes may be providing database services,
+others web applications, and others may do a mixture. However, the underlying hosts themselves are configured to be
+OpenShift nodes and are configured identical to each other.
+
+To create a brand new cluster:
+
+.. note::
+
+    If you did not create a ContainerManager you can omit ``--container-manager``.
+
+.. include:: examples/commctl_create_cluster.rst
+
+Adding Hosts To The Cluster
+---------------------------
+
+Adding new hosts to Commissaire comes in two forms. Automatic registration and manual additions.
+
+Automatic Registration
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+    This will be streamlined next release.
+
+This is the preferred way to deal with new hosts. This method allows Commissaire and, if configured, the
+ContainerManager, to be ready to use the host.
+
+Before we start you will need `make-mime.py`_ script from cloud-init's ``tools`` directory.
+
+First, you will need to create a configuration file for ``cloud-config``. Generally you'll do this once
+and then change the ``cluster`` key for different clusters. Here is an example:
+
+.. literalinclude:: ../tools/cloud-init/commissaire.txt
+   :language: shell
+
+Next, copy you will need a ``cloud-config`` text file. Unless you have a custom one created already
+the following will work:
+
+.. literalinclude:: ../tools/cloud-init/cloud-config.txt
+   :language: yaml
+
+Now copy ``tools/cloud-init/part-handler.py`` into the same directory as``commissaire.txt`` and ``cloud-config.txt``.
+
+Then, use `make-mime.py`_ to put it all together into a ``user-data`` file we can use in cloud providers.
+
+.. _make-mime.py: http://bazaar.launchpad.net/~cloud-init-dev/cloud-init/trunk/view/head:/tools/make-mime.py
+
+.. note::
+
+    You may see "text/x-commissaire-host' for attachment 3 may be incorrect!". Don't worry! This is OK.
+
+.. code-block:: shell
+
+   $ python make-mime.py \
+            --attach cloud-config.txt:cloud-config \
+            --attach part-handler.py:part-handler \
+            --attach commissaire.txt:x-commissaire-host \
+            > user-data
+
+Now provide the new ``user-data`` file when provisioning new hosts in your cloud provider. When the new host starts
+it will automatically register into Commissaire.
+
+Manual Registration
+~~~~~~~~~~~~~~~~~~~
+You can also add hosts into Commissaire in a manual fashion. To do this you will need:
+
+1. The host to have ``sshd`` running
+2. The host to have ``sshd`` port open
+3. The private key to an administrative user on the host (EG: ``root``)
+
+Let's say you have a host called ``192.168.152.110`` which you'd like to add to the cluster ``my_cluster``. You also
```

Comment:
The rewording of this paragraph is a great improvement and is good enough on its own. My only suggestions would be to add wording that describes where in EC2 and GCE (and Azure if you're super ambitious :-) ) the user would have put the public part of the key into the Cloud Provider's respective IAM consoles. That would help a lot of users who are just getting comfortable with working in a cloud setting. The other suggestion I would have is to add a boilerplate "How to create an ssh keypair" for users who are possibly testing this out locally via libvirt/vagrant/etc. However, these things are certainly secondary now and could be added later if wanted, since your wording is now understandable to even beginning users at this point in my opinion.